### PR TITLE
[major]: Add MongoDB vector store option

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: app/api/ingestPdf/route.ts [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-const pinecone = new Pinecone({
-  apiKey: process.env.PINECONE_API_KEY ?? '',
-});
-
-if (!process.env.PINECONE_INDEX_NAME) {
-  throw new Error('Missing Pinecone index name in .env file');
-}
+import { loadEmbeddingsModel } from '../utils/embeddings';
+import { loadVectorStore } from '../utils/vector_store';
+import { MongoClient } from 'mongodb';
```
Comment (jacoblee93) [inline app/api/ingestPdf/route.ts:8]: nit: Use as a type import

File: package.json [hunk 2]
```diff
@@ -55,5 +58,8 @@
     "prisma": "^5.2.0",
     "tsx": "^3.12.3",
     "typescript": "^5.2.2"
+  },
+  "overrides": {
+    "mongodb": "^6.3.0"
```
Comment (jacoblee93) [inline package.json:63]: Why do we need this?

File: app/api/chat/route.ts [hunk 3]
```diff
@@ -139,8 +137,8 @@ export async function POST(req: NextRequest) {
     // "Pick" the answer from the retrieval chain output object and stream it as bytes.
     const outputChain = RunnableSequence.from([
       conversationalRetrievalChain,
-      new RunnablePick({ keys: "answer" }),
-      new HttpResponseOutputParser({ contentType: "text/plain" }),
+      new RunnablePick({ keys: 'answer' }),
```
Comment (jacoblee93) [inline app/api/chat/route.ts:140]: If you wouldn't mind - `conversationalRetrievalChain.pick('answer')` would be a little nicer here

File: package.json (hunk 2 above)
Comment (bracesproul) [reply package.json:63]: don't. removed

**Pull request closed without merging**